pkg/common/decoder.go

```diff
@@ -28,6 +28,11 @@ func ReadIntX(v string, p int) int {
 	return int(v[p+3])*0x1000000 + int(v[p+2])<<16 + int(v[p+1])<<8 + int(v[p])
 }
 
+// ReadBufferShort reads a short value from a byte array at position p.
```

Comment:
is this needed? where is it used?

Reply:
yup, I added this earlier before switching to binary.Read.  Good catch!